source/aggregation/pipeline-stages.txt

```diff
@@ -0,0 +1,318 @@
+===============
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: node.js, code example, transform, pipeline
+   :description: Learn the different possible stages of the aggregation pipeline in the {+driver-short+}.
+
+Overview
+------------
+
+On this page, you can learn how to create an aggregation pipeline and pipeline
+stages by using methods in the {+driver-short+}.
+
+Build an Aggregation Pipeline
+-----------------------------
+
+You can use the {+driver-short+} to build an aggregation pipeline by creating a
+pipeline variable or passing aggregation stages directly into the aggregation
+method. See the following examples to learn more about each of these approaches.
+
+.. tabs::
+
+  .. tab:: Create a Pipeline
+    :tabid: pipeline-definition
+
+      .. code-block:: javascript
+
+        // Defines the aggregation pipeline
+        const pipeline = [
+          { $match: { ... } },
+          { $group: { ... } }
+        ];
+
+        // Executes the aggregation pipeline
+        const results = collection.aggregate(pipeline);
+
+  .. tab:: Direct Aggregation
+    :tabid: pipeline-direct
+
+    .. code-block:: javascript
+
+      // Defines and executes the aggregation pipeline
+      collection.aggregate([
+        { $match: { ... } },
+        { $group: { ... } }
+      ]);
+
+
+Aggregation Stage Methods
+-------------------------
+
+The following table lists the stages in the aggregation pipeline. To learn more
+about an aggregation stage and see a code example in a Node.js application,
+follow the link from the stage name to its reference page in the {+mdb-server+}
+manual.
+
+
+.. list-table::
+  :header-rows: 1
+  :widths: 30 70
+
+  * - Stage
+    - Description
+
+  * - :manual:`$addFields </reference/operator/aggregation/addFields/>`
+    - Adds new fields to documents. Outputs documents that contain both the
+      existing fields from the input documents and the newly added fields.
+      
+      ``$set`` is an alias for ``$addFields``.
+
+  * - :manual:`$bucket </reference/operator/aggregation/bucket/>`
+    - Categorizes incoming documents into groups, called buckets,
+      based on a specified expression and bucket boundaries.
+
+  * - :manual:`$bucketAuto </reference/operator/aggregation/bucketAuto/>`
+    - Categorizes incoming documents into a specific number of
+      groups, called buckets, based on a specified expression.
+      Bucket boundaries are automatically determined in an attempt
+      to evenly distribute the documents into the specified number
+      of buckets.
+
+  * - :manual:`$changeStream </reference/operator/aggregation/changeStream/>`
+    - Returns a change stream cursor for the collection. 
+      
+      Instead of being passed to the ``.aggregate()`` method,
+      ``$changeStream`` uses the ``.watch()`` method on a ``collection``
+      object.
```

Comment:
S: (applies everywhere) remove periods from method references
```suggestion
      Instead of being passed to the ``aggregate()`` method,
      ``$changeStream`` uses the ``watch()`` method on a ``Collection``
      object.
```